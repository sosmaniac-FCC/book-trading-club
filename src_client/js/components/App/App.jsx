import React from "react";

import Copyright from '../Copyright/Copyright.jsx';
import Navigation from '../Navigation/Navigation.jsx';
import Routes from '../Routes/Routes.jsx';

class App extends React.Component {
    constructor(props) {
        super(props);
    }
    
    render() {
        return (
            <div className="container">
                <header className="container">
                    <Navigation/>
                </header>
                <main className="container">
                    <Routes/>
                </main>
                <footer className="container page-footer purple accent-1">
                    <Copyright/>
                </footer>
            </div>
        );
    }
};

export default App;