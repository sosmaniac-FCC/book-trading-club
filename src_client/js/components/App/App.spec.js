// expect from jest_expect.js
/* global expect */

import React from 'react';
import { shallow, configure } from 'enzyme';
import enzymeAdapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';

configure({ adapter: new enzymeAdapter() });

import App from './App.jsx';

test('Top-level application layout loads as expected.', () => {
    const component = shallow(<App />);
    const tree = toJSON(component);
    expect(tree).toMatchSnapshot();
});

test('Top-level application layout should have no style attributes.', () => {
    const component = shallow(<App />);
    expect(component.find('[style]')).toHaveLength(0);
});