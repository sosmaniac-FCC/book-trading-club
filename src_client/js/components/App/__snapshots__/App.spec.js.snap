// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Top-level application layout loads as expected. 1`] = `
<div
  className="container"
>
  <header
    className="container"
  >
    <Navigation />
  </header>
  <main
    className="container"
  >
    <Routes />
  </main>
  <footer
    className="container page-footer purple accent-1"
  >
    <Copyright />
  </footer>
</div>
`;
